code/modules/antagonists/clock_cult/mobs/eminence.dm

```diff
@@ -146,10 +184,37 @@
 	else
 		to_chat(user, "<span class='warning'>You're already on the station!</span>")
 
+//=====Teleport to servant=====
+/obj/effect/proc_holder/spell/targeted/eminence/servant_warp
+	name = "Jump to Servant"
+	desc = "Teleport yourself to a specific servant."
+	action_icon_state = "Spatial Warp"
+
+/obj/effect/proc_holder/spell/targeted/eminence/servant_warp/cast(list/targets, mob/user)
+	//Get a list of all servants
+	var/choice = input(user, "Select servant", "Warp to...", null) in GLOB.all_servants_of_ratvar
```

Comment:
What does all_servants_of_ratvar hold? Can't you just get mob from there instead of looping it again?

Reply:
It holds mobs. Using input returns a string which is the name if atoms are provided